apps/token-manager/app/src/utils.js

```diff
@@ -10,3 +12,76 @@ export function formatBalance(amount, base) {
 
   return `${whole}${fraction === '0' ? '' : `.${fraction.slice(0, 2)}`}`
 }
+
+// Calculates and returns stakes as percentages, adding a “rest” percentage for
+// values that are not included.
+//
+// Params:
+//   - amounts: (BN.js array) the amounts to be converted in percentages.
+//   - total: (BN.js) the total amount.
+//   - maxIncluded: (Number) the max count of items to include in the result.
+//
+// Returns an array of objects where:
+//   - `index` is the original index in `amounts`, or -1 if it’s the “rest”.
+//   - `amount` is the original amount provided.
+//   - `percentage` is the calculated percentage.
+//
+export function stakesPercentages(amounts, total, maxIncluded) {
+  const hasRest = amounts.length > maxIncluded
+
+  // percentage + two digits (only to sort them by closest to the next integer)
+  const pctPrecision = new BN(10000)
+
+  // Calculate the percentages of all the stakes
+  const stakes = amounts
+    .map((amount, index) => ({ index, amount }))
+    .filter(({ amount }) => !amount.isZero())
+    .map(stake => ({
+      ...stake,
+      percentage: stake.amount.mul(pctPrecision).div(total),
```

Comment:
If I remember correctly, the reason I did the calculations with native numbers originally was that I hit problems with this type of arithmetic with BNs. Give it a go with http://aragon.aragonpm.com/#/4color.aragonid.eth/